src/main/java/teammates/ui/webapi/FeedbackSessionPublishedEmailWorkerAction.java

```diff
@@ -19,20 +20,36 @@ public JsonResult execute() {
         String feedbackSessionName = getNonNullRequestParamValue(ParamsNames.FEEDBACK_SESSION_NAME);
         String courseId = getNonNullRequestParamValue(ParamsNames.COURSE_ID);
 
-        FeedbackSessionAttributes session = logic.getFeedbackSession(feedbackSessionName, courseId);
+        if (!isCourseMigrated(courseId)) {
+            FeedbackSessionAttributes session = logic.getFeedbackSession(feedbackSessionName, courseId);
+            if (session == null) {
+                log.severe("Feedback session object for feedback session name: " + feedbackSessionName
+                        + " for course: " + courseId + " could not be fetched.");
+                return new JsonResult("Failure");
+            }
+            List<EmailWrapper> emailsToBeSent = emailGenerator.generateFeedbackSessionPublishedEmails(session);
+            try {
+                taskQueuer.scheduleEmailsForSending(emailsToBeSent);
+                logic.updateFeedbackSession(
+                        FeedbackSessionAttributes
+                                .updateOptionsBuilder(session.getFeedbackSessionName(), session.getCourseId())
+                                .withSentPublishedEmail(true)
```

Comment:
Apologies, we might need sentPublishedEmail in the entity model as well. Sessions can be published manually or automatically at a certain time, no way to differentiate them at the moment.

Could you help make the change in another PR?

Reply:
see #12248